ssmlib/backends/multipath.py

```diff
@@ -102,9 +104,12 @@ def get_volume_data(self, volname):
             data['dev_size'] = misc.get_device_size(data['dev_name'])
             data['nodes'] = []
             data['total_nodes'] = 0
-            for swap in misc.get_swaps():
-                if swap[0] == data['dev_name']:
-                    data['mount'] = "SWAP"
+            if data['dev_name'] in self.mounts:
+                data['mount'] = self.mounts[data['dev_name']]['mp']
+            else:
+                for swap in misc.get_swaps():
+                    if swap[0] == data['dev_name']:
+                        data['mount'] = "SWAP"
```

Comment:
get_volume_data is called for every volume and so it would make much more sense to get_swaps in init (as you do with get_mounts) and save the results.

Reply:
sure, thanks lukas, i will check ASAP

Reply:
Hi Lukas, sorry for the wait.  The tests have been added now.  Please review at your convenience.  Thanks!